backends/bmv2/common/helpers.h

```diff
@@ -62,18 +59,44 @@ class V1ModelProperties {
     static const cstring validField;
 };
 
-using ErrorValue = unsigned int;
-using ErrorCodesMap = ordered_map<const IR::IDeclaration *, ErrorValue>;
+// XXX(hanw): This convenience class stores pointers to the data structures
+// that are commonly used during the program translation. Due to the limitation
+// of current IR structure, these data structure are only refreshed by the
+// evaluator pass. In the long term, integrating these data structures as part
+// of the IR tree would simplify this kind of bookkeeping effort.
```

Comment:
+1

Reply:
This may not be possible, since some of these data structures may have cycles, and the IR prohibits cycles.

Reply:
I also think that the IR is complicated enough, we should not carry within the IR stuff that we only need occasionally.